mantidproject/mantid
# CCache on windows

**Description of work.**
This PR ensures ccache is configured on windows when building with Ninja. It should hopefully mean far quicker build speeds when modifying certain files.

**To test:**
1. Update your environment to make sure ccache is installed. You can do `mamba env update -f mantid-developer-win.yml --prune`
1. Starting with an empty build folder, create a ninja build by first opening `x64 Native Tools command prompt for VS 2019`, activating the `mantid-developer` conda env and running `cmake --preset=win-ninja` from the source directory, and `ninja` from the build directory.
2. Change a commonly used header file like `Instrument.h`
3. Do ninja again in the build directory, 
4. Change `Instrument.h` back to its preview form. Do ninja again from the build directory and it should build fairly quickly
4. Check your `AppData/Roaming` directory for a `ccache` folder, to check that caching is being done
5. Create an MSVC build with `cmake --preset=win-msvc -B=buildmsvc`. Build it using the visual studio IDE and check that debugging still works (i.e. you can still git breakpoints)

Part of #34490

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
